includes/Services/LabkiPageRegistry.php

```diff
@@ -265,6 +265,39 @@ private function stripNamespaces( array $titles ): array {
         }
         return $out;
     }
+
+    public function getRewriteMapForRepo( int $repoId ): array {
+        $dbr = MediaWikiServices::getInstance()->getDBLoadBalancer()->getConnection( DB_REPLICA );
+    
+        $res = $dbr->newSelectQueryBuilder()
+            ->select( [ 'lp.name', 'lp.final_title' ] )
+            ->from( self::TABLE, 'lp' )
+            ->join( 'labki_pack', 'pack', [ 'lp.pack_id = pack.pack_id' ] )
+            ->where( [ 'pack.content_repo_id' => $repoId ] )
+            ->distinct()
+            ->caller( __METHOD__ )
+            ->fetchResultSet();
+    
+        if ( !$res->numRows() ) {
+            wfDebugLog( 'Labki', "No rewrite map rows found for repo_id=$repoId" );
+            return [];
+        }
+    
+        $map = [];
+        foreach ( $res as $row ) {
+            $orig = str_replace( ' ', '_', (string)$row->name );
```

Comment:
makes sense to return a complete rewritemap from this method, i would remove the duplicated cleaning operations from the other method